airflow/cli/commands/dag_command.py

```diff
@@ -375,15 +375,17 @@ def dag_list_dags(args, session=NEW_SESSION) -> None:
             file=sys.stderr,
         )
 
-    def get_dag_detail(dag: DAG) -> dict:
+    def get_dag_detail(dag: DAG) -> dict | None:
         dag_model = DagModel.get_dagmodel(dag.dag_id, session=session)
+        if not dag_model:
+            return None
```

Comment:
If the `dag_model` is not available, can we use `dag` details? This would ensure backwards compatibility

Reply:
yeah, but wondering if that would be correct data for the end users?

Reply:
Yes, it will be. we just need to return the details there before the change that broke this. See https://github.com/apache/airflow/pull/35250/files#diff-91a3a78f5c49452359633888b8bea004a0d230e75e72cc08f03dcce0d159d4b8L368-L373

Reply:
cool, then I'll change it. thanks!

Reply:
Also look at the custom columns stuff before the change

Reply:
@ephraimbuddy I have tried to use the dag some fields are not available in dag and only available in dagmodel for them I'm setting columns value as none. PTAL